### PR TITLE
Fix/2521 slotted side nav visible when changing expanded

### PR DESCRIPTION
**Pull request by GCHQ-Developer-847**
## Summary of the changes
Fix side nav slotted nav item bug by ensuring correct styling is executed when externally controlled.

I also added a delay to tooltips on the side nav navigation items to prevent them showing too soon when the nav items are used for toggling expanded state. I thought I may as well fix it as part of this PR as it was a very obvious issue - saves opening a new ticket for it

**How to test:** Go to the "React Router" story and click on the "Home (expanded toggle)" nav item, which changes the expanded state of the side nav. The bug mentioned on the ticket should no longer be occurring

## Related issue
#2521 

### General 

- [x] Changes to docs package checked and committed.
- [x] All acceptance criteria reviewed and met. 

### Testing

- [x] Relevant unit tests and visual regression tests added. 
- [x] Visual testing against Figma component specification completed. 
- [x] Playground stories in React Storybook up to date, with any prop changes and additions addressed. 
- [x] Compare performance of modified components against develop using Performance addon in React Storybook.

### Accessibility 

- [x] Accessibility Insights FastPass performed.
- [x] A11y unit test added and yields no issues.
- [x] A11y plug-in on Storybook yields no issues. 
- [x] Manual screen reader testing performed using NVDA and VoiceOver. 
- [x] Manual keyboard testing for keyboard controls and logical focus order. 
- [x] Correct roles used and ARIA attributes used correctly where required. 
- [x] Logical heading structure is maintained, and the HTML elements used for headings can be changed to fit within the wider page structure. 

### Resize/zoom behaviour 

- [x] Page can be zoomed to 400% with no loss of content. 
- [x] Screen magnifier used with no issues. 
- [x] Text resized to 200% with no loss of content.
- [x] Text spacing increased as per the [WCAG 1.4.12 success criterion](https://www.w3.org/TR/WCAG21/#text-spacing) with no loss of content.

### System modes

- [x] Browser setting 'prefers reduced motion' tested. No animations or motion visible whilst this setting is on. 
- [x] Windows High Contrast mode tested with no loss of content.
- [x] System light and dark mode tested with no loss of content.
- [x] Browser support tested (Chrome, Safari, Firefox and Edge). 

### Testing content extremes

- [x] Min/max content examples tested with no loss of content or overflow. 
- [x] All prop combinations work without issue. 
- [x] Tested for FOUC (Flash of Unstyled Content) in both SSR (Server-Side Rendering) and SSG (Static Site Generation) settings.
- [x] Controlled and uncontrolled input components tested.
- [x] Props/slots can be updated after initial render.

### CODE REVIEW

**Comment by GCHQ-Developer-847** [comment]
This is the behaviour which was happening with the tooltips for which I added a fix

https://github.com/user-attachments/assets/875f2113-a0a8-413a-99e4-5b0150ca35fb

File: packages/react/src/component-tests/IcSideNavigation/IcSideNavigation.cy.tsx [hunk 1]
```diff
@@ -348,14 +349,31 @@ describe("IcSideNavigation", () => {
...
+    it("should render expanded when expanded state is externally controlled", () => {
       mount(<DynamicExpandedSideNav />);
 
       cy.get("#expand-btn").click();
+
+      cy.checkA11yWithWait(undefined, 500);
       cy.checkSideNavSize(true);
+      cy.compareSnapshot({
+        name: "controlled-expanded",
+        testThreshold: setThresholdBasedOnEnv(DEFAULT_TEST_THRESHOLD + 0.024),
```
Comment (evenstensberg) [inline packages/react/src/component-tests/IcSideNavigation/IcSideNavigation.cy.tsx:361]: pr looks good. I have a question. Do you have a policy for threshold, or is mainly adjusted to pass the CI with the minimal threshold possible?

**Review by GCHQ-Developer-299 — APPROVED** [review]

File: packages/react/src/component-tests/IcSideNavigation/IcSideNavigation.cy.tsx (hunk 1 above)
Comment (GCHQ-Developer-847) [reply packages/react/src/component-tests/IcSideNavigation/IcSideNavigation.cy.tsx:361]: The [guidance in our CONTRIBUTING.md file](https://github.com/mi6/ic-ui-kit/blob/main/CONTRIBUTING.md#generatingupdating-cypress-screenshots) explains our process for the visual regression tests. We basically generate the screenshots on the CI and then run the tests locally against those screenshots and set the threshold as low as possible (0.01% above the difference).

**Comment by GCHQ-Developer-741** [comment]
tooltip_sn

I see two issues but I don't know if they're related to this. The first is that the tooltip hovers away from the side nav when clicking to collapse it, occurring by clicking the nav-item on the icon so that the mouse stays over the side nav in both expanded and collapsed states.

Second is the extra focus indicator shown in the image (This happens on all of them when clicked, probs a seperate ticket but if its an easy fix that could go in this then I suppose thats a bonus?)

**Review by GCHQ-Developer-299 — APPROVED** [review]

**Review by GCHQ-Developer-741 — APPROVED** [review]

**Pull request merged**